src/kudu/consensus/leader_election.cc

```diff
@@ -139,56 +153,55 @@ bool VoteCounter::AreAllVotesIn() const {
   return GetTotalVotesCounted() == num_voters_;
 }
 
-void FlexibleVoteCounter::FetchTopologyInfo(
-    std::map<std::string, int>* voter_distribution,
-    std::map<std::string, int>* le_quorum_requirement) {
-  CHECK(voter_distribution);
-  CHECK(le_quorum_requirement);
-  voter_distribution->clear();
-  le_quorum_requirement->clear();
+PotentialNextLeaderResponse::PotentialNextLeaderResponse(
+    PotentialNextLeaderResponse::Status s) {
+  status = s;
+  potential_leader_uuid = "";
+  potential_leader_term = -1;
+}
+
+PotentialNextLeaderResponse::PotentialNextLeaderResponse(
+    PotentialNextLeaderResponse::Status s, const std::string& uuid,
+    const int64_t term) {
+  status = s;
+  potential_leader_uuid = uuid;
+  potential_leader_term = term;
+}
 
+void FlexibleVoteCounter::FetchTopologyInfo() {
   // Step 1: Compute number of voters in each region.
   // Also, computes the leader's region from its UUID.
-  std::string leader_region = "";
   GetRegionalCountsFromConfig(
-      config_, last_known_leader_.uuid, voter_distribution,
-      &leader_region);
+      config_, last_known_leader_.uuid, &voter_distribution_,
+      &last_known_leader_region_);
```

Comment:
there is no guarantee that last_known_leader_region_ is returned. It could be empty.